### PR TITLE
Interpreter/Jit64: Emulate FMA accurately in more cases

### PR DESCRIPTION
**Pull request by JosJuice**
Thought my efforts to improve cross-architecture netplay sync would only involve making JitArm64 more accurate? Nope! It's time to make Jit64 more accurate now. (And the interpreter too, as a bonus.)

### CODE REVIEW

**Comment by Sintendo** [comment]
Commit message for the Jit64 commit appears to have been truncated.

**Comment by JosJuice** [comment]
Fixed.

File: Source/Core/Core/PowerPC/Interpreter/Interpreter_FPUtils.h [hunk 1]
```diff
@@ -268,7 +268,7 @@ inline FPResult NI_madd(UReg_FPSCR* fpscr, double a, double c, double b)
     return result;
   }
 
-  result.value += b;
+  result.value = std::fma(a, c, b);
```
Comment (degasus) [inline Source/Core/Core/PowerPC/Interpreter/Interpreter_FPUtils.h:271]: Is there still a point in checking if `a*c` is a NaN? Especially as the `isnan(b)` handling is in there as well?

**Review by Tilka — APPROVED** [review]
lgtm, but please make sure this has been tested before merging

**Comment by JosJuice** [comment]
I've tested that F-Zero GX runs correctly with all the Jit64 codepaths (hardware FMA, software FMA, no FMA).

**Pull request merged**